components/ws-daemon/nsinsider/main.go

```diff
@@ -251,6 +252,162 @@ func main() {
+					vethAddr := net.IPNet{
+						IP:   net.IPv4(10, 0, 5, 1),
+						Mask: mask,
+					}
+					cethAddr := net.IPNet{
+						IP:   net.IPv4(10, 0, 5, 2),
+						Mask: mask,
+					}
+					masqueradeAddr := net.IPNet{
+						IP:   vethAddr.IP.Mask(mask),
+						Mask: mask,
+					}
+					ipCmd, iptablesCmd := "/usr/sbin/ip", "/usr/sbin/iptables"
+					netns := "workspace-ns"
+
+					cmd := exec.Command(ipCmd, "link", "add", vethIf, "type", "veth", "peer", "name", cethIf)
+					out, err := cmd.CombinedOutput()
+					if err != nil {
+						return xerrors.Errorf("create a veth pair (%v) failed: %q\n%v",
+							cmd.Args,
+							string(out),
+							err,
+						)
+					}
+
+					path := "/var/run/netns"
+					if err := os.MkdirAll(path, 0755); err != nil {
+						return xerrors.Errorf("create a dir %s failed: %v", path, err)
+					}
+					if err := os.Symlink(fmt.Sprintf("/proc/%d/ns/net", c.Int("target-pid")), filepath.Join(path, netns)); err != nil {
+						return xerrors.Errorf("create a symlink to netns (%v) failed: %q\n%v",
+							cmd.Args,
+							string(out),
+							err,
+						)
+					}
+					cmd = exec.Command(ipCmd, "link", "set", cethIf, "netns", netns)
+					out, err = cmd.CombinedOutput()
+					if err != nil {
+						return xerrors.Errorf("link cethIf to netns (%v) failed: %q\n%v",
+							cmd.Args,
+							string(out),
+							err,
+						)
+					}
+
+					cmd = exec.Command(ipCmd, "addr", "add", vethAddr.String(), "dev", vethIf)
+					out, err = cmd.CombinedOutput()
+					if err != nil {
+						return xerrors.Errorf("assign IP address to the vethIf (%v) failed: %q\n%v",
+							cmd.Args,
+							string(out),
+							err,
+						)
+					}
+
+					cmd = exec.Command(ipCmd, "netns", "exec", netns, "ip", "addr", "add", cethAddr.String(), "dev", cethIf)
+					out, err = cmd.CombinedOutput()
+					if err != nil {
+						return xerrors.Errorf("assign IP address to the cethIf (%v) failed: %q\n%v",
+							cmd.Args,
+							string(out),
+							err,
+						)
+					}
+
+					cmd = exec.Command(ipCmd, "link", "set", vethIf, "up")
+					out, err = cmd.CombinedOutput()
+					if err != nil {
+						return xerrors.Errorf("bring up the vethIf (%v) failed: %q\n%v",
+							cmd.Args,
+							string(out),
+							err,
+						)
+					}
+
+					cmd = exec.Command(ipCmd, "netns", "exec", netns, "ip", "link", "set", cethIf, "up")
+					out, err = cmd.CombinedOutput()
+					if err != nil {
+						return xerrors.Errorf("bring up the cethIf (%v) failed: %q\n%v",
+							cmd.Args,
+							string(out),
+							err,
+						)
+					}
+
+					cmd = exec.Command(ipCmd, "netns", "exec", netns, "ip", "link", "set", "lo", "up")
+					out, err = cmd.CombinedOutput()
+					if err != nil {
+						return xerrors.Errorf("bring up the lo (%v) failed: %q\n%v",
+							cmd.Args,
+							string(out),
+							err,
+						)
+					}
+
+					cmd = exec.Command(iptablesCmd, "-A", "FORWARD", "-o", containerIf, "-i", vethIf, "-j", "ACCEPT")
```

Comment:
Why are you adding these FORWARD rules? The default policy is ACCEPT
```
$ sudo iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination 
```

Reply:
That's right. We don't need it. Before this code, I tried to add rules to the original chain and I forget removing the codes.